src/data/nat/choose.lean

```diff
@@ -9,13 +9,23 @@ open nat
 
 open_locale big_operators
 
-lemma nat.prime.dvd_choose {p k : ℕ} (hk : 0 < k) (hkp : k < p) (hp : prime p) : p ∣ choose p k :=
-have h₁ : p ∣ fact p, from hp.dvd_fact.2 (le_refl _),
-have h₂ : ¬p ∣ fact k, from mt hp.dvd_fact.1 (not_le_of_gt hkp),
-have h₃ : ¬p ∣ fact (p - k), from mt hp.dvd_fact.1 (not_le_of_gt (nat.sub_lt_self hp.pos hk)),
-by rw [← choose_mul_fact_mul_fact (le_of_lt hkp), mul_assoc, hp.dvd_mul, hp.dvd_mul] at h₁;
+lemma nat.prime.dvd_choose {p a b : ℕ} (hap : a < p) (hbp : b < p) (h : p ≤ a + b)
```

Comment:
This looks good to me, but I would call this `dvd_choose_add`, and revert the other lemma to `dvd_choose`.
@rwbarton What do you think?

Reply:
I was going to suggest `dvd_choose_self` for the other lemma before seeing that this PR already chose that name too, so that one seems like a good name. Maybe this one should indeed be `dvd_choose_add`, I could see a case to have a third lemma without "`a + b`" (but with extra side conditions) that reduces to this one.